compiler/plugins/input/Torch/torch-iree/PluginRegistration.cpp

```diff
@@ -72,6 +75,28 @@ struct TorchSession
     typeMnemonics.insert("tm_tensor");
     typeMnemonics.insert("torch");
   }
+
+  std::string detectCustomInputOperations(ModuleOp &module) override {
```

Comment:
Is a string sufficient though? Also is their a hierarchy here? E.g., if you see torch_c and torch, then return torch?

Reply:
It's sufficient for calling `detectCustomInputOperations` -> the existing `extendCustomInputConversionPassPipeline` method (since the latter takes a "typeMnemonic" string). There is a chance here to further rework the API though. It wasn't quite clear to me what would be needed by the API until I had this plumbed all the way across... I'll think on this a bit.

Reply:
A string is easier to manage across arbitrary plugins, but we could add more structure to that string if needed (namespaces, parameters/modifiers, etc.). We could also have a list/map of strings for combinations, but at that point we may just want to allow all plugins to do their input preprocessing in an arbitrary order (one after another), at least until we find a case of plugins conflicting with one another.

Regarding the multiple strings here: "torch" and "torch_c" are both covered by the "torch" pipeline. "tm_tensor" is a legacy pipeline that we might just be able to remove entirely now. I'm not too concerned with doing a priority selection of one or the other.